sumpy/tools.py

```diff
@@ -862,4 +911,21 @@ def matvec_toeplitz_upper_triangular(first_row, vector):
         output[row] = sym.Add(*terms)
     return output
 
+
+to_complex_dtype_dict = {
+    np.complex64: np.complex64,
+    np.complex128: np.complex128,
+    np.float32: np.complex64,
+    np.float64: np.complex128,
+}
+
+
+def to_complex_dtype(dtype):
+    try:
```

Comment:
```suggestion
    dtype = np.dtype(dtype).type
    try:
```

Reply:
Done in 0c59fc4538d1d956b659a748745888800620d785